BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -407,21 +408,27 @@ extension ScrollView {
 
     static func calculateContentInset(
         scrollViewInsets: UIEdgeInsets,
+        contentInsets: UIEdgeInsets,
         adjustedContentInsets: UIEdgeInsets,
         keyboardBottomInset: CGFloat,
         refreshControlState: PullToRefreshBehavior,
         refreshControlBounds: CGRect?
     ) -> UIEdgeInsets {
         var finalContentInset = scrollViewInsets
 
+        // The safe area may not actually be applied to the scroll view depending on
+        // the content height and inset adjustment mode. We need to determine the actual applied
+        // safe area insets by subtracting the content inset from the adjusted content inset.
+        let appliedSafeAreaInset = adjustedContentInsets - contentInsets
```

Comment:
Do we also need to check the `contentInsetAdjustmentBehavior`; or is this right in all cases?

Reply:
This should (finally) be right in all cases; since the `adjustedContentInset` will be adjusted based on the `contentInsetAdjustmentBehavior`

Reply:
👍 great!